relentless/simulate/lammps.py

```diff
@@ -0,0 +1,361 @@
+            cmds += ['create_atoms {typeid} random {N} {seed} box'.format(typeid=sim.type_map[i],
+                                                                          N=sim.ensemble.N[i],
+                                                                          seed=self.seed+sim.type_map[i]-1)]
+        cmds += ['mass * 1.0',
+                 'velocity all create {temp} {seed}'.format(temp=sim.ensemble.T,
+                                                            seed=self.seed)]
+
+        cmds += self.attach_potentials(sim)
+
+        return cmds
+
+class MinimizeEnergy(LAMMPSOperation):
+    def __init__(self, energy_tolerance, force_tolerance, max_iterations, dt):
+        self.energy_tolerance = energy_tolerance
+        self.force_tolerance = force_tolerance
+        self.max_iterations = max_iterations
+
+    def to_commands(self, sim):
+        cmds = ['minimize {etol} {ftol} {maxiter} {maxeval}'.format(etol=self.energy_tolerance,
+                                                                    ftol=self.force_tolerance,
+                                                                    maxiter=self.max_iterations,
+                                                                    maxeval=100*self.max_iterations)]
+
+        return cmds
+
+class AddLangevinIntegrator(LAMMPSOperation):
+    def __init__(self, dt, friction, seed):
+        self.friction = friction
+        self.seed = seed
+
+        self._fix_langevin = self.add_fix()
+        self._fix_nve = self.add_fix()
+
+    def to_commands(self, sim):
+        # obtain per-type mass
+        mass_per_type = {}
+
+        mass_ptr = sim.lammps.extract_atom('mass',2)
+        Ntypes = len(sim.ensemble.types)
+        mass = np.ctypeslib.as_array(mass_ptr,(Ntypes+1,))[1:]
+
+        types_ptr = sim.lammps.extract_atom('type',0)
+        Natoms = sum(sim.ensemble.N.todict().values())
+        types = np.unique(np.ctypeslib.as_array(types_ptr,(Natoms,)))
+        types = [str(t) for t in types]
+
+        for t,m in zip(types,mass):
+            mass_per_type[t] = m
+
+        # obtain per-type friction factor
+        if np.isscalar(self.friction):
+            friction_per_type = {}
+            for t in types:
+                friction_per_type[t] = self.friction
+        else:
+            if sorted(self.friction.keys()) != sorted(types):
+                return ValueError('The friction factor types must match the atom types.')
+            friction_per_type = self.friction
+
+        # compute per-type damping parameter
+        damp_per_type = {}
+        for t in friction_per_type:
+            damp_per_type[t] = mass_per_type[t]/friction_per_type[t]
+
+        damp_ref = list(damp_per_type.values())[0]
+        scale_str = ''
+        for t in damp_per_type:
+            if damp_per_type[t] == damp_ref:
+                continue
+            scale_str += (str(t) + ' ' + str(damp_per_type[t]/damp_ref) + ' ')
+        scale_str.rstrip()
```

Comment:
I think this will achieve what you're trying to do more compactly. I also repeated the "scale" keyword, since I think you need it for each type? Note the `i+1` is important to index the types correctly with enumerate.
```suggestion
        damp_ref = damp[1]
        scale = damp / damp_ref
        scale_str = ' '.join(['scale {} {}'.format(i+1,s) for i,s in enumerate(scale[1:])])
```